spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonDeserializer.java

```diff
@@ -136,15 +148,15 @@ public void setUseTypeMapperForKey(boolean isKey) {
 	public void configure(Map<String, ?> configs, boolean isKey) {
 		setUseTypeMapperForKey(isKey);
 		try {
-			if (isKey && configs.containsKey(DEFAULT_KEY_TYPE)) {
-				if (configs.get(DEFAULT_KEY_TYPE) instanceof Class) {
-					this.targetType = (Class<T>) configs.get(DEFAULT_KEY_TYPE);
+			if (isKey && configs.containsKey(KEY_DEFAULT_TYPE)) {
+				if (configs.get(KEY_DEFAULT_TYPE) instanceof Class) {
+					this.targetType = (Class<T>) configs.get(KEY_DEFAULT_TYPE);
 				}
-				else if (configs.get(DEFAULT_KEY_TYPE) instanceof String) {
-					this.targetType = (Class<T>) ClassUtils.forName((String) configs.get(DEFAULT_KEY_TYPE), null);
+				else if (configs.get(KEY_DEFAULT_TYPE) instanceof String) {
+					this.targetType = (Class<T>) ClassUtils.forName((String) configs.get(KEY_DEFAULT_TYPE), null);
 				}
 				else {
-					throw new IllegalStateException(DEFAULT_KEY_TYPE + " must be Class or String");
+					throw new IllegalStateException(KEY_DEFAULT_TYPE + " must be Class or String");
 				}
 			}
 			else if (!isKey && configs.containsKey(DEFAULT_VALUE_TYPE)) {
```

Comment:
This causes a deprecation warning.

Also there is a `KafkaMessageListenerContainerTests` which uses constant as well with similar deprecation warning. Should not use it any more.

Please, fix these concerns, run `gradlew clean check`. 

And also, please, take a look into the Reference Manual (`.adoc` files) if we use these constant there. And, therefore, fix the doc on the matter.

Otherwise looks good.

:+1:

Reply:
keep the old DEFAULT_VALUE_TYPE here because of the compatibility, I think we can't remove it in next patch version, but can be removed in next major version, how about adding a `@SuppressWarnings("deprecation")` on the method?

Reply:
Right, we will remove that code in the next major (or maybe even minor) version. However I’m not a fan of `@SuppressWarnings("deprecation")`. Would be better to use literals here instead of deprecated constants. However we will need TODO comment here do not forget to remove the obsolete code